nnstreamer_example/custom_example_opencv/nnstreamer_customfilter_opencv_scaler.cc

```diff
@@ -0,0 +1,171 @@
+    if (strv[0] != NULL) {
+      data->out_width = (uint32_t) g_ascii_strtoll (strv[0], NULL, 10);
+    } else {
+      data->out_width = 0;
+    }
+    if (strv[1] != NULL) {
+      data->out_height = (uint32_t) g_ascii_strtoll (strv[1], NULL, 10);
+    } else {
+      data->out_height = 0;
+    }
+    g_strfreev (strv);
+  }
+
+  return data;
+}
+
+/**
+ * @brief exit callback of tensor_filter custom
+ */
+static void
+pt_exit (void *private_data, const GstTensorFilterProperties * prop)
+{
+  pt_data *pdata = static_cast<pt_data *> (private_data);
+  assert (pdata);
+  g_free (pdata);
+}
+
+/**
+ * @brief setInputDimension callback of tensor_filter custom
+ */
+static int
+set_inputDim (void *private_data, const GstTensorFilterProperties * prop,
+    const GstTensorsInfo * in_info, GstTensorsInfo * out_info)
+{
+  pt_data *pdata = static_cast<pt_data *> (private_data);
+
+  assert (pdata);
+  assert (in_info);
+  assert (out_info);
+
+  out_info->num_tensors = 1;
+
+  for (int i = 0; i < NNS_TENSOR_RANK_LIMIT; i++)
+    out_info->info[0].dimension[i] = in_info->info[0].dimension[i];
+
+  /* Save width and height of an input tensor */
+  pdata->in_width = in_info->info[0].dimension[1];
+  pdata->in_height = in_info->info[0].dimension[2];
+
+  /* Update output dimension [1] and [2] with new-x, new-y */
+  if (pdata->out_width > 0)
+    out_info->info[0].dimension[1] = pdata->out_width;
+  if (pdata->out_height > 0)
+    out_info->info[0].dimension[2] = pdata->out_height;
+
+  out_info->info[0].type = in_info->info[0].type;
+  return 0;
+}
+
+/**
+ * @brief invoke-alloc callback of tensor_filter custom
+ */
+static int
+pt_allocate_invoke (void *private_data,
+    const GstTensorFilterProperties * prop, const GstTensorMemory * input,
+    GstTensorMemory * output)
+{
+  pt_data *pdata = static_cast<pt_data *> (private_data);
+  size_t in_size, out_size;
+  void* buffer;
+  cv::Mat img_src, img_dst;
+
+  assert (pdata);
+  assert (input);
+  assert (output);
+
+  in_size = gst_tensor_info_get_size (&prop->input_meta.info[0]);
+  out_size = gst_tensor_info_get_size (&prop->output_meta.info[0]);
+  buffer = g_malloc (in_size);
+  output[0].data = g_malloc (out_size);
+
+  /* Get Mat object from input tensor */
+  memcpy (buffer, input[0].data, in_size);
+  img_src = cv::Mat (pdata->in_height, pdata->in_width, CV_8UC3, buffer);
+  cv::cvtColor (img_src, img_src, CV_BGR2RGB);
+
+  /* Scale from the shape of input tensor to that of output tensor
+   * which is given as custom property */
+  cv::resize (img_src, img_dst, cv::Size(pdata->out_width, pdata->out_height), 
+            0, 0, CV_INTER_NN);
+
+  /* Convert Mat object to output tensor */
+  cv::cvtColor (img_dst, img_dst, CV_RGB2BGR);
+  memcpy (output[0].data, img_dst.data, out_size);
+
+  g_free(buffer);
+
+  return 0;
+}
+
+/**
+ * @brief tensor_filter custom subplugin definition
+ */
+static NNStreamer_custom_class NNStreamer_custom_body = {
+  .allocate_outbuf_in_invoke = 0,
```

Comment:
Please remove this line, allocate_outbuf_in_invoke = 0

Reply:
Basically I agree with your opinion since it is **NOT elegant**. However, what if this line is removed, then below error occurs.
```bash
nnstreamer_customfilter_opencv_scaler.cc:168:1: sorry, unimplemented: non-trivial designated initializers not supported
```